tools/fuzzer/main.cc

```diff
@@ -41,8 +41,99 @@
 
 using bazel::tools::cpp::runfiles::Runfiles;
 
-constexpr char SOURCE_PATH_KEY[] = "lldb_eval/testdata/fuzzer_binary.cc";
-constexpr char BINARY_PATH_KEY[] = "lldb_eval/testdata/fuzzer_binary";
+static constexpr char SOURCE_PATH_KEY[] = "lldb_eval/testdata/fuzzer_binary.cc";
+static constexpr char BINARY_PATH_KEY[] = "lldb_eval/testdata/fuzzer_binary";
+
+enum Verbosity {
+  ShowMismatchesOrErrors,
+  ShowEverything,
+};
+
+void eval_and_print_expr(lldb::SBFrame& frame, const std::string& expr,
+                         Verbosity verbosity) {
+  auto lldb_value = frame.EvaluateExpression(expr.c_str());
+  auto lldb_err = lldb_value.GetError();
+
+  lldb::SBError lldb_eval_err;
+  auto lldb_eval_value =
+      lldb_eval::EvaluateExpression(frame, expr.c_str(), lldb_eval_err);
+
+  bool value_mismatch;
+  if (lldb_value.GetValue() != nullptr &&
+      lldb_eval_value.GetValue() != nullptr) {
+    value_mismatch =
+        strcmp(lldb_value.GetValue(), lldb_eval_value.GetValue()) != 0;
+  } else {
+    // Pointer comparison: Mismatch if one value is null and the other is not
+    value_mismatch = lldb_value.GetValue() != lldb_eval_value.GetValue();
+  }
+
+  bool type_mismatch;
+  if (lldb_value.GetTypeName() != nullptr &&
+      lldb_eval_value.GetTypeName() != nullptr) {
+    type_mismatch =
+        strcmp(lldb_value.GetTypeName(), lldb_eval_value.GetTypeName()) != 0;
+  } else {
+    // Pointer comparison: Mismatch if one type is null and the other is not
```

Comment:
Not 100%, but I've never seen `GetTypeName` to return `nullptr`

Reply:
Yep, had a look at the lldb source code and it's using [https://lldb.llvm.org/cpp_reference/ConstString_8h_source.html](ConstString) under the hood, which doesn't look like it could be null (since the default ctor initializes it to an empty string).

Reply:
Nope, I spotted a case where lldb-eval returned a value with an empty type, so the extra test will have to stay for now.